crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1356,6 +1356,14 @@ impl<'db> Type<'db> {
                 }
             }
 
+            // If we instantiate a subclass of typing.Any, then we infer the result to be an
+            // instance of the Any type.
+            Type::ClassLiteral(ClassLiteralType { class })
+                if class.is_subclass_of_base(db, ClassBase::Any) =>
+            {
+                CallOutcome::callable(Type::Any)
+            }
+
```

Comment:
It's awesome that you found the right spot for this and figured out how to do it, but (as described above) I don't think this is quite the behavior we'll want.

Reply:
Reverted